File: questions/804/java/Solution.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        if (words.length>100){
            return 0;
        }
        String[] morseDict = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        HashSet<String> morseSet = new HashSet<String>();
        for(String word : words){
            if (word.length() >12){
                return 0;
            }
            StringBuilder sb = new StringBuilder();
            for (char c: word.toCharArray()){
                sb.append(morseDict[c-'a']);
            }
            morseSet.add(sb.toString());
        }
        return morseSet.size();
    }
}

File: questions/804/python/Solution.py
class Solution(object):
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        morseMap = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        morseSet = set()
        for word in words:
            morseCode = []
            for c in word:
                morseCode.append(morseMap[ord(c)-97])
            morseSet.add(''.join(morseCode))
        return len(morseSet)